packages/store/src/StoreCore.sol

```diff
@@ -583,9 +583,6 @@ library StoreCore {
    * @param fieldLayout The field layout for the record.
    */
   function deleteRecord(ResourceId tableId, bytes32[] memory keyTuple, FieldLayout fieldLayout) internal {
-    // Emit event to notify indexers
-    emit Store_DeleteRecord(tableId, keyTuple);
-
```

Comment:
we also need to move the early return for offchain tables below the event emission, otherwise there is no event emitted for offchain tables (and the onBefore hook is not triggered)

Reply:
can we add a test for this to make sure we don't regress?

Reply:
Added a couple tests